import prefer `js` over `css/sass/scss`

It appears that 7.0 breaks sass modules we are using from npm. I guess it is related to https://github.com/webpack-contrib/sass-loader/pull/479

For example foundation-sites that we are using imports it's internal modules like this:
```
// Sass utilities
@import 'util/util';
```

Now, unfortunately 'util/util' is a valid file inside node_modules (from 'node-util' pkg), and the build fails with errror like this:

```
ERROR in ./src/style.scss (./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/style.scss)
Module build failed: 
// Copyright Joyent, Inc. and other Node contributors.
^
      Invalid CSS after "...N THE SOFTWARE.": expected 1 selector or at-rule, was "var formatRegExp = "
      in ###/webpack-sass-broken-imports/node_modules/util/util.js (line 1, column 1)
 @ ./src/style.scss 2:14-117
 @ ./src/index.js
```
repro repo: https://github.com/larpo/webpack-sass-broken-module-imports

Is there a way to revert to the old behaviour? I don't see any mentions on READMe about how to configure the aliasing behaviour.

Team, 

 When i tried to install the following npm sass fix

   npm install git://github.com/webpack-contrib/sass-loader.git#fix-resolve-conflict --save-dev

getting the following error 
   npm ERR! error: pathspec 'fix-resolve-conflict' did not match any file(s) known to git.

Please provide any comments/suggestions

Thanks

--- a/lib/loader.js
+++ b/lib/loader.js
@@ -12,6 +12,23 @@ const normalizeOptions = require('./normalizeOptions');
 
 let nodeSassJobQueue = null;
 
+// Very hacky check
+function hasGetResolve(loaderContext) {
+  return (
+    loaderContext.getResolve &&
+    // eslint-disable-next-line no-underscore-dangle
+    loaderContext._compiler &&
+    // eslint-disable-next-line no-underscore-dangle
+    loaderContext._compiler.resolverFactory &&
+    // eslint-disable-next-line no-underscore-dangle
+    loaderContext._compiler.resolverFactory._create &&
+    /cachedCleverMerge/.test(
+      // eslint-disable-next-line no-underscore-dangle
+      loaderContext._compiler.resolverFactory._create.toString()
+    )
+  );
+}
+
 /**
  * The sass-loader makes node-sass and dart-sass available to webpack modules.
  *
@@ -37,11 +54,11 @@ function sassLoader(content) {
 
   let resolve = pify(this.resolve);
 
-  // Supported since v4.27.0
-  if (this.getResolve) {
+  // Supported since v4.36.0
+  if (hasGetResolve(self)) {
     resolve = this.getResolve({
-      mainFields: ['sass', 'main'],
-      extensions: ['.scss', '.sass', '.css'],
+      mainFields: ['sass', 'style', '...'],
+      extensions: ['.scss', '.sass', '.css', '...'],
     });
   }
 

